lib/executor/constant_arrival_rate.go

```diff
@@ -0,0 +1,270 @@
+func NewConstantArrivalRateConfig(name string) ConstantArrivalRateConfig {
+	return ConstantArrivalRateConfig{
+		BaseConfig: NewBaseConfig(name, constantArrivalRateType),
+		TimeUnit:   types.NewNullDuration(1*time.Second, false),
+	}
+}
+
+// Make sure we implement the lib.ExecutorConfig interface
+var _ lib.ExecutorConfig = &ConstantArrivalRateConfig{}
+
+// GetPreAllocatedVUs is just a helper method that returns the scaled pre-allocated VUs.
+func (carc ConstantArrivalRateConfig) GetPreAllocatedVUs(es *lib.ExecutionSegment) int64 {
+	return es.Scale(carc.PreAllocatedVUs.Int64)
+}
+
+// GetMaxVUs is just a helper method that returns the scaled max VUs.
+func (carc ConstantArrivalRateConfig) GetMaxVUs(es *lib.ExecutionSegment) int64 {
+	return es.Scale(carc.MaxVUs.Int64)
+}
+
+// GetDescription returns a human-readable description of the executor options
+func (carc ConstantArrivalRateConfig) GetDescription(es *lib.ExecutionSegment) string {
+	preAllocatedVUs, maxVUs := carc.GetPreAllocatedVUs(es), carc.GetMaxVUs(es)
+	maxVUsRange := fmt.Sprintf("maxVUs: %d", preAllocatedVUs)
+	if maxVUs > preAllocatedVUs {
+		maxVUsRange += fmt.Sprintf("-%d", maxVUs)
+	}
+
+	timeUnit := time.Duration(carc.TimeUnit.Duration)
+	arrRate := getScaledArrivalRate(es, carc.Rate.Int64, timeUnit)
+	arrRatePerSec, _ := getArrivalRatePerSec(arrRate).Float64()
+
+	return fmt.Sprintf("%.2f iterations/s for %s%s", arrRatePerSec, carc.Duration.Duration,
+		carc.getBaseInfo(maxVUsRange))
+}
+
+// Validate makes sure all options are configured and valid
+func (carc ConstantArrivalRateConfig) Validate() []error {
+	errors := carc.BaseConfig.Validate()
+	if !carc.Rate.Valid {
+		errors = append(errors, fmt.Errorf("the iteration rate isn't specified"))
+	} else if carc.Rate.Int64 <= 0 {
+		errors = append(errors, fmt.Errorf("the iteration rate should be more than 0"))
+	}
+
+	if time.Duration(carc.TimeUnit.Duration) <= 0 {
+		errors = append(errors, fmt.Errorf("the timeUnit should be more than 0"))
+	}
+
+	if !carc.Duration.Valid {
+		errors = append(errors, fmt.Errorf("the duration is unspecified"))
+	} else if time.Duration(carc.Duration.Duration) < minDuration {
+		errors = append(errors, fmt.Errorf(
+			"the duration should be at least %s, but is %s", minDuration, carc.Duration,
+		))
+	}
+
+	if !carc.PreAllocatedVUs.Valid {
+		errors = append(errors, fmt.Errorf("the number of preAllocatedVUs isn't specified"))
+	} else if carc.PreAllocatedVUs.Int64 < 0 {
+		errors = append(errors, fmt.Errorf("the number of preAllocatedVUs shouldn't be negative"))
+	}
+
+	if !carc.MaxVUs.Valid {
+		errors = append(errors, fmt.Errorf("the number of maxVUs isn't specified"))
+	} else if carc.MaxVUs.Int64 < carc.PreAllocatedVUs.Int64 {
+		errors = append(errors, fmt.Errorf("maxVUs shouldn't be less than preAllocatedVUs"))
+	}
+
+	return errors
+}
+
+// GetExecutionRequirements just reserves the number of specified VUs for the
```

Comment:
Minor: this method doesn't actually *reserve* anything, right? "Reserve" has some functional implications this does not do, AFAICS.

Consider rewriting this and the other executor's `GetExecutionRequirements()` comments with something along the lines of "... returns the execution plan ...".

Also, nitpick, but we should probably standardize the usage of "execution plan" / "execution requirements", or do you think these two as being distinct?

Reply:
So, is this resolved? :smile: Please elaborate if it's not, I'm just trying to clean up all outstanding comments in the PR...

Reply:
@na-- I thought you would update the comment to remove "reserves"(?).

Reply:
:man_facepalming: yeah, my bad, sorry! I saw your :+1: and didn't bother rereading the whole discussion :blush: Will fix it later today

Reply:
Resolved by https://github.com/loadimpact/k6/commit/3421078d34e9eb2dce507c2687eeb3981d1afc87